cpu/samd21/periph/uart.c

```diff
@@ -100,41 +119,81 @@ int uart_init_blocking(uart_t uart, uint32_t baudrate)
-
-            /* set to LSB, asynchronous mode without parity, PAD0 Tx, PAD1 Rx,
-             * 16x over-sampling, internal clk */
-            UART_0_DEV.CTRLA.reg = SERCOM_USART_CTRLA_DORD \
-                                    | SERCOM_USART_CTRLA_FORM(0x0) \
-                                    | SERCOM_USART_CTRLA_SAMPA(0x0) \
-                                    | SERCOM_USART_CTRLA_TXPO(0x0) \
-                                    | SERCOM_USART_CTRLA_RXPO(0x1) \
-                                    | SERCOM_USART_CTRLA_SAMPR(0x0) \
-                                    | SERCOM_USART_CTRLA_MODE_USART_INT_CLK;
-
-            /* Set baud rate */
-            UART_0_DEV.BAUD.bit.BAUD = baud_calculated;
-
-            /* enable receiver and transmitter, one stop bit*/
-            UART_0_DEV.CTRLB.reg = (SERCOM_USART_CTRLB_RXEN | SERCOM_USART_CTRLB_TXEN);
-            while(UART_0_DEV.SYNCBUSY.bit.CTRLB);
+            break;
+#endif
+#if UART_1_EN
+        case UART_1:
+            port_group = &UART_1_PORT;
+            uart_dev = &UART_1_DEV;
+            uart_ref_f = UART_1_REF_F;
+            tx_pin = UART_1_TX_PIN;
+            rx_pin = UART_1_RX_PIN;
+            pins = UART_1_PINS;
+
+            /* Turn on power manager for sercom5 */
+            PM->APBCMASK.reg |= PM_APBCMASK_SERCOM5;
+
+            /* configure GCLK0 to feed sercom5 */;
+            GCLK->CLKCTRL.reg = (uint16_t)((GCLK_CLKCTRL_CLKEN | GCLK_CLKCTRL_GEN_GCLK0 | (SERCOM5_GCLK_ID_CORE << GCLK_CLKCTRL_ID_Pos)));
+            while (GCLK->STATUS.bit.SYNCBUSY);
+
+            GCLK->CLKCTRL.reg = (uint16_t)((GCLK_CLKCTRL_CLKEN | GCLK_CLKCTRL_GEN_GCLK0 | (SERCOM5_GCLK_ID_SLOW << GCLK_CLKCTRL_ID_Pos)));
+            while (GCLK->STATUS.bit.SYNCBUSY);
 
             break;
 #endif
+        default:
+            return -2;
     }
 
+    if (port_group == NULL) {
+        return -2;
+    }
+
+    /* configure PINS to input/output*/
+    port_group->DIRSET.reg = (1 << tx_pin);  /* tx's direction is output */
+
+    port_group->PINCFG[rx_pin % 32].bit.DRVSTR = 0; /* pin drive strength is set to normal */
+    port_group->PINCFG[tx_pin % 32].bit.DRVSTR = 0;
+    port_group->PINCFG[rx_pin % 32].bit.INEN = 1; /* buffer rx pin's value */
+    port_group->PINCFG[rx_pin % 32].bit.PMUXEN = 1; /* enables peripheral multiplexer */
+    port_group->PINCFG[tx_pin % 32].bit.PMUXEN = 1; /* enables peripheral multiplexer */
+
+
+    /* enable PMUX for pins and set to config D. See spec p. 12 */
+    port_group->WRCONFIG.reg = PORT_WRCONFIG_WRPINCFG     /* PINCFGy registers of the selected pins will be updated */
+                                | PORT_WRCONFIG_WRPMUX    /* PMUXn registers of the selected pins will be updated */
+                                | PORT_WRCONFIG_PMUX(0x3) /* Multiplexer peripheral function D */
+                                | PORT_WRCONFIG_PMUXEN    /* Peripheral multiplexer enable */
+                                | pins;                   /* Pinmask */
```

Comment:
You need to select the **upper** 16 pins of PORT for pins 22 and 23 by setting the `HWSEL` bit in this register write.
See section 21.8.11 p. 389.